src/components/Checklist.js

```diff
@@ -1,48 +1,73 @@
 import React from 'react'
+import { ChecklistHeader,
+         ChecklistTitle,
+         ChecklistToolbar,
+         ChecklistItems,
+         ChecklistAddItemForm,
+         ChecklistFooter } from './'
+import { addArrayItem, generateId } from '../helpers/helpers'
 
 export class Checklist extends React.Component {
+
+  constructor () {
+    super()
+
+    this.state = {
+      items: [
+        { id: 1, name: 'Item One', isCompleted: false },
+        { id: 2, name: 'Item Two', isCompleted: false },
+        { id: 3, name: 'Item Three', isCompleted: true }
+      ],
+      newItem: ''
+    }
+
+    this.handleInputChange = this.handleInputChange.bind(this)
+    this.handleSubmit = this.handleSubmit.bind(this)
+  }
+
+  handleInputChange (e) {
+    this.setState({
+      newItem: e.target.value
+    })
+  }
+
+  handleSubmit (e) {
+    e.preventDefault()
+    const newItem = {
+      id: generateId(),
+      name: this.state.newItem,
+      isCompleted: false
+    }
+    if (!this.state.newItem) return
+    this.setState({
+      items: addArrayItem(newItem, this.state.items),
```

Comment:
I think it's less confusing to just write `items: [...this.state.items, newItem]` right here instead of making a helper method.
Or, `items: this.state.items.concat(newItem)`. It's just a style choice.